.github/workflows/pr.yml

```diff
@@ -240,11 +242,74 @@ jobs:
         run: |
           echo "VEDA_CONFIG_REPO=${{ vars.VEDA_CONFIG_REPO_ORG }}/${{ vars.VEDA_CONFIG_REPO_NAME }}" >> $GITHUB_ENV
 
+
+      - name: Get Installation ID
+        id: get-installation-id
+        run: |
+          echo "${{ secrets.APP_PEM }}" > app.pem
+
+          APP_ID="${{ vars.VEDA_CONFIG_APP_ID }}"
+
+          # Ensure APP_ID is set
+          if [ -z "$APP_ID" ]; then
+          echo "Error: APP_ID is not set!"
+          exit 1
+          fi
+
+          # Ensure private key exists
+          if [ ! -f app.pem ]; then
+          echo "Error: Private key (app.pem) not found!"
+          exit 1
+          fi
+
+          # Generate JWT
+          HEADER='{"alg":"RS256","typ":"JWT"}'
+          PAYLOAD='{"iat":'$(date +%s)',"exp":'$(($(date +%s)+540))',"iss":'"$APP_ID"'}'
+
+          # Base64 URL-safe encode header and payload
+          BASE64_HEADER=$(printf '%s' "$HEADER" | openssl base64 -A | tr -d '=' | tr '/+' '_-')
+          BASE64_PAYLOAD=$(printf '%s' "$PAYLOAD" | openssl base64 -A | tr -d '=' | tr '/+' '_-')
+
+          # Combine Header and Payload
+          JWT_HEADER_PAYLOAD="${BASE64_HEADER}.${BASE64_PAYLOAD}"
+          echo "Header.Payload: $JWT_HEADER_PAYLOAD"
+
+          JWT_SIGNATURE=$(printf '%s' "$JWT_HEADER_PAYLOAD" | openssl dgst -sha256 -sign app.pem | openssl base64 -A | tr -d '=' | tr '/+' '_-')
+
+          if [ -z "$JWT_SIGNATURE" ]; then
+              echo "Error: Failed to create JWT_SIGNATURE!"
+              exit 1
+          fi
+          JWT="$JWT_HEADER_PAYLOAD.$JWT_SIGNATURE"
+
+          # Get installation ID
+          RESPONSE=$(curl -s -X GET \
+            -H "Authorization: Bearer $JWT" \
+            -H "Accept: application/vnd.github+json" \
+            https://api.github.com/app/installations)
+
+          INSTALLATION_ID=$(echo $RESPONSE | jq -r '.[0].id')
+          echo "installation_id=$INSTALLATION_ID" >> $GITHUB_ENV
+          TOKEN_RESPONSE=$(curl -s -X POST \
+            -H "Authorization: Bearer $JWT" \
+            -H "Accept: application/vnd.github+json" \
+            https://api.github.com/app/installations/$INSTALLATION_ID/access_tokens)
+
+          INSTALLATION_TOKEN=$(echo $TOKEN_RESPONSE | jq -r '.token')
+          echo "installation_token=$INSTALLATION_TOKEN" >> $GITHUB_ENV
+
       - name: Clone veda-config repository
         run: |
-          git clone https://github.com/${{ env.VEDA_CONFIG_REPO }}.git
+          git clone https://x-access-token:${{ env.installation_token }}@github.com/${{ env.VEDA_CONFIG_REPO}}.git
           ls
 
+      - name: Authenticate with GitHub App
+        run: |
+          echo "${{ env.installation_token }}" | gh auth login --with-token
+          git config --global user.name "VEDA-Github-Actor[bot]]"
+          git config --global user.email 192275278+VEDA-Github-Actor[bot]@users.noreply.github.com
```

Comment:
https://api.github.com/users/VEDA-Github-Actor[bot] this is where I got info on the email (I used the ID). I'm not sure if this is something we want to store in an env var?

Reply:
I think it makes sense to use an environment variable but this is OK here for now, too

Reply:
https://github.com/NASA-IMPACT/veda-data/pull/223/commits/ca2de70a6f6936171c71bb9d16777cd1fac34e07 and variable `GH_ACTOR_EMAIL ` populated as well !